subprojects/packagefiles/htslib/test/meson.build

```diff
@@ -0,0 +1,119 @@
+# tests rely on -fvisibility=hidden symbols
+htslib_test_lib = static_library(
+  'statichts', [
+    htslib_config_h,
+    htslib_config_vars_h,
+    htslib_version_h],
+  objects: htslib_lib.extract_all_objects(recursive : false),
+  include_directories: '..',
+  install : false,
+  dependencies : [
+    libm_dep,
+    thread_dep,
+    zlib_dep])
+
+test('hts_endian',
+  executable(
+    'hts_endian', [
+      htslib_config_h,
+      files(['hts_endian.c'])],
+    link_with : htslib_test_lib,
+    include_directories: '..',
+    install : false))
+
+test('test_expr',
+  executable(
+    'test_expr', [
+      htslib_config_h,
+      files(['test_expr.c'])],
+    link_with : htslib_test_lib,
+    include_directories: '..',
+    install : false))
+
+test('test_kfunc',
+  executable(
+    'test_kfunc', [
+      htslib_config_h,
+      files(['test_kfunc.c'])],
+    link_with : htslib_test_lib,
+    include_directories: '..',
+    install : false))
+
+test('test_kstring',
+  executable(
+    'test_kstring', [
+      htslib_config_h,
+      files(['test_kstring.c'])],
+    link_with : htslib_test_lib,
+    include_directories: '..',
+    install : false))
+
+test('test_str2int',
+  executable(
+    'test_str2int', [
+      htslib_config_h,
+      files(['test_str2int.c'])],
+    link_with : htslib_test_lib,
+    include_directories: '..',
+    install : false))
+
+test('test_fieldarith',
+  executable(
+    'fieldarith', [
+      htslib_config_h,
+      files(['fieldarith.c'])],
+    link_with : htslib_test_lib,
+    include_directories: '..',
+    install : false),
+  args : files(['fieldarith.sam']))
+
+test('test_hfile',
+  executable(
+    'test_hfile', [
+      htslib_config_h,
+      files(['hfile.c'])],
+    link_with : htslib_test_lib,
+    include_directories: '..',
+    install : false),
+  workdir : meson.project_source_root())
```

Comment:
FWIW you could get away with a lower version of meson by using join_paths(meson.current_source_dir(), '..') or by assigning current_source_dir from the parent meson.build and reusing that variable. 0.48 is a lot more widespread, especially on stable distros (Debian buster, Ubuntu 20.04 primarily) and it seems fairly easy to do...